robot/lib/Inc/drivers/GenericRadio.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+
+class GenericRadio {
+public:
+    /**
+     * Blocking call to send X number of bytes from `data` over the radio
+     * 
+     * @param data raw array of data to send over the radio
+     * @param numBytes number of bytes in the array to send
+     * 
+     * @return number of bytes sent
+     * 
+     * @note It is assumed that `data` is at least X bytes long
+     */
+    virtual unsigned int send(const uint8_t* data, const unsigned int numBytes) = 0;
```

Comment:
Why do we need dynamic dispatch?

We should be explicitly minimizing use of `new` in firmware, and dynamic dispatch encourages people to do that. And we _really_ don't need it here IMO.

Reply:
This was just an easy way to abstract out the direct radio driver from everything else. It's very annoying when we swap devices every other year and nobody actually removes and replaces the radio code in a clean way to fit with a specific device.

My hope with this was to enforce an API such that it's not a problem (given it's not async). It provides a clear line between what should be device specific code and what should be application code. In my mind, that clear separation is more useful than the overhead this creates.

Reply:
That's fair. My point was more that I don't want to encourage people to dynamically allocate things that could be static. As long as we avoid that I'm fine.